persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxTreeGenerator.java

```diff
@@ -1158,6 +1179,28 @@ public static void generateRelations(ArrayList<Entity> entityArray) {
                             for (Integer index : indexesToRemove) {
                                 childEntity.getFields().remove(index.intValue());
                             }
+                            if (childEntity.getRelations().get(relationIndex).getKeyColumns().isEmpty() &&
+                                    childEntity.getRelations().get(relationIndex).getReferences().isEmpty()) {
+                                for (String key : entity.getKeys()) {
+                                    String modifiedKey = key.replace("\"", "").trim();
+                                    childEntity.getRelations().get(relationIndex).getKeyColumns().add(
+                                            entity.getEntityName().toLowerCase() + modifiedKey.substring(0, 1)
+                                                    .toUpperCase() + modifiedKey.substring(1));
+                                    childEntity.getRelations().get(relationIndex).getReferences().add(modifiedKey);
+                                }
+                            } else if (childEntity.getRelations().get(relationIndex).getKeyColumns().isEmpty()) {
+                                for (String key : entity.getKeys()) {
+                                    String modifiedKey = key.replace("\"", "").trim();
+                                    childEntity.getRelations().get(relationIndex).getKeyColumns().add(
+                                            entity.getEntityName().toLowerCase() + modifiedKey.substring(0, 1)
+                                                    .toUpperCase() + modifiedKey.substring(1));
+                                }
+                            } else if (childEntity.getRelations().get(relationIndex).getReferences().isEmpty()) {
+                                for (String key : entity.getKeys()) {
+                                    String modifiedKey = key.replace("\"", "").trim();
+                                    childEntity.getRelations().get(relationIndex).getReferences().add(modifiedKey);
+                                }
+                            }
```

Comment:
don't we need a else block here?

Reply:
Else state will be the case where both annotations are present and this case is the default case where there is no additional processing is required. As we change other states (annotations are empty and half empty) in to this case through the if and else if blocks, the Else block is not required for this instance.